### DISCUSSION TITLE
How to access the record in afterCreate()?

### DISCUSSION THREAD

**Discussion by vv-ac** [filamentphp/filament#2431] [Help] [answered]
Hi all.

after creating a new instance of a resource model I want to create another model that belongsTo this model.
Thefore I would like to use the id of the newly created model in creating the new one.
How can I access this attribute?
I tried to simply use $this->record but this did not help.

Any idea?
Thanks!

**Comment by danharrin** [comment]
`$this->record` is definitely right, `dd($this->record)` in the hook to double check it it's a Model object.

**Reply by vv-ac** [reply]
Returned null.

After that I switched the code it bit using this in ManageDocuments.php:
```
<?php

namespace App\Filament\Resources\DocumentResource\Pages;

use App\Models\Document;

use Illuminate\Database\Eloquent\Model;

use App\Filament\Resources\DocumentResource;
use Filament\Resources\Pages\ManageRecords;

class ManageDocuments extends ManageRecords
{
    protected static string $resource = DocumentResource::class;

    protected static string $model = Document::class;

    public $record;

    public function mount(Document $record)
    {
        $this->record = $record;
    }

    protected function afterCreate()
    {
        return dd($this->record);

        // ...
    }
}

```

Now returning in:
```
Declaration of App\Filament\Resources\DocumentResource\Pages\ManageDocuments::mount(App\Models\Document $record) must be compatible with Filament\Resources\Pages\ListRecords::mount(): void
```

**Reply by tnatanael** [reply]
> `$this->record` is definitely right, `dd($this->record)` in the hook to double check it it's a Model object.

That is the right answer for 3x

**Comment by Z3d0X** [comment, marked as answer]
In manageRecords, you can use 
`$this->getMountedTableActionRecord()` in `afterCreate()`

**Reply by vv-ac** [reply]
Hi, thanks for replying to my question. Can you explaing a bite more in detail where/ how to apply that to access the record?
What is "mountesTableAction" referring to here?

Thanks!

**Reply by danharrin** [reply]
"mounted table action" is the current active table action that is open

"record" is the Eloquent record that is associated with that table action

`$this->getMountedTableActionRecord()` gets the Eloquent model object for the active table action

**Reply by vv-ac** [reply]
Thank you both, that works!
